app/models/User.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.mvc.{Request, WrappedRequest}
+
+case class User[T](mtditid: String, arn: Option[String])(implicit request: Request[T]) extends WrappedRequest[T](request) {
+  def isAgent: Boolean = arn.nonEmpty
+}
```

Comment:
needs to be tested 
<img width="1340" alt="Screenshot 2020-10-29 at 2 19 18 pm" src="https://user-images.githubusercontent.com/63247344/97586210-e84b6500-19f1-11eb-80be-441d09b21d69.png">